### PR TITLE
Added mm to units and color output

### PR DESCRIPTION
**Pull request by baxerus**
I added 'mm' as supported units. For me it's the most interesting one actually.
But this has to be tested more! Especially files with '%' as units. In my opinion, this did not work right beforehand and I tried to fix it now.

I also added color output (with opacity). Very cool 😊

I added my svg test files. But it would be good, to add more (especially with different units).

### CODE REVIEW

**Comment by martymcguire** [comment]
Hey! Thanks for all of this, and my apologies for not replying before now.

I haven't had much time to contribute to this project, recently and unfortunately I don't see that changing in the near future.

This all looks great, but there are several things going on at once.

I would be happy to merge the coding style-changes as suggested by PyCharm.

I love the idea of test files - how would you feel about creating some instructions for how to run the test files and verify that they work correctly?

If you're confident that `mm` units are working, I'd love to merge that in. If you're not sure about `%` units I am more hesitant to have the code "support" those units if the output might be unexpected.

The color updates look really cool, but are kind of a whole separate feature.

Would you be willing to rebase these updates into a separate PR?

* PyCharm-related code style changes
* mm unit updates
* test files (w/ instructions)

And then separate these into their own PRs?

*  `%` unit support (until it is confirmed working)
* color updates (I think this is only missing some docs about how to use it in practice)

Once again, thanks so much for these updates!

**Comment by baxerus** [comment]
Already created a new pull request #6 for only the PyCharm or general refactoring because this was easy. For the rest I have to struggle through git rebases... I will try tomorrow or later for the rest (first testings destroyed nearly the whole commit tree. Hard git learning curve 😃).

Test files:
I actually simply compared the Inkscape view with the OpenSCAD view. And also compared measurements for mm files.
But I needed files with % as units (I think Inkscape can't produce them nowadays, at least I wasn't able to do so. But of course you could do outside of Inkscape. My testifies are handwritten).

% unit support:
For my two test files the % output worked. The 50% file is half as high as the 100% file, like in Inkscape (btw the github preview doesn't do this 😊). The underlying dimensions of the path aren't preserved in the moment (500px width ends as 500mm width in OpenScad), this is something that could be improved. But we should discuss and/or define what the actual wanted behavior is here (I think it's hard to tell for a file that should be 100% of it's parent element).

color updates:
actually the color updates simply look for parseable (aka simple) color definitions in the SVG file and if there are some adds the OpenSCAD color modifier to the "test" output (see my changes to the output creation). It's not integrated into the modules by design, because in most cases you would want to extrude the 2D polygons and than you lose the color modification. The color has to be at the beginning of a subtree in OpenSCAD. So with the "test" output in the way it is now a user could simply copy paste the color modifier to where he needs if (if he wants to).
At the end the color updates are simply a good looking feature. It allows to faster compare the Inkscape and OpenSCAD views. But I think it isn't a high priority thing, because in the end you lose the color information, when OpenSCAD renders the STL (damn OpenSCAD 😊).

**Comment by baxerus** [comment]
Sorry had problems with my internet connection (and not much time). Perhaps I will have time to generate the other pull requests this weekend.

**Comment by martymcguire** [comment]
No problem and no hurry! Looking forward to them.

**Comment by baxerus** [comment]
I still didn't find the time to reorder my commits and pack it into separate branches and pull requests.
But I used the plugin (with my modifications) for some time now (great for see creating throught pictures) and haven't had issues so far.

**Comment by baxerus** [comment]
I've done more testing (and created some really nice printed pictures with Inkscape from photos). Everything is tested as much as I can and is working.
You could simply merge the pull request (from my, the developers 😊, side).

Also I tried to separate the changes into separate branches afterwards and always failed (that's why I started testing instead 👍).

**Comment by martymcguire** [comment]
Hey baxerus,

Thanks for your patience, and thanks for staying on me case about this!

I'm prepared to accept the changes and do the merge, but could I ask you for a couple of changes, first?

1. Can you document the new features in the README file so folks understand what to expect from them?
2. Can you add yourself to the contributor's list in the copyright section of the files that you have updated? I want to make sure you are properly credited. :}

Thanks again!!

**Comment by baxerus** [comment]
1. Of course I can update the README, with a new example and description. No problem.
2. Most of the files do not have a Copyright section yet. And I couldn't find one with a contributor list.

So for both, how should I do it? Simply change all files to the best of my knowledge and make a new pull request (that would be depending on this one than)? Or do you want to add the Copyright sections with (empty) contributor lists into all files (and perhaps a Contributor section at the end of the README?) first? To make sure everything is convenient before I do my changes... 😉

**Comment by martymcguire** [comment]
Re: README - great!

Re: contributor list - I was thinking of having you add your preferred attribution to the README (changing the Author section to Contributors and listing both of us), as well as having you add a Copyright (c) 2016 [your preferred attribution here] in the files you changed, like `openscad-poly.py`.

Currently everything in this repo is covered by the GPLv2 license (per LICENSE). If you would like to cover your test SVGs differently, feel free to add a section in the repo.

TL;DR, if you write up the example usages in the README, feel free to just tell me your preferred attribution (name and email, name and website, pseudonym and bitcoin address, whatever) and I'll figure out some good places to list us. :}

**Comment by baxerus** [comment]
I added contributor sections in the README and in every source files (you want to add yourself there after the merge, I didn't know how you like your copyright attributions 😉).
I also added a little bit of description in the README (and deleted ToDos, yeah 👍 ).

To the license of the SVG. For me GPLv2 is fine, but one of the files is not created by myself, but downloaded from wikimedia (look at the commit message of that file). Wikimedia says it is public domain, but I am not sure if (and how) it is compatible with GPLv2. Perhaps this is a problem and it's easier to remove the file, perhaps there is no problem at all. I simply do not know.

**Comment by martymcguire** [comment]
Thanks for all the updates!

If a file is released into the public domain, it is available to be distributed here (or anywhere!). I think it is worth mentioning in the README that that particular file is public domain, along with an attribution link back to the page it came from.

If you can drop that note in the README, I'll get my own attribution info in place and we'll get this merged!

**Comment by baxerus** [comment]
Added the attribution and link to the Wikimedia. Do you need more?

**Comment by martymcguire** [comment]
Thanks so much for your patience.

Everything is now merged and hopefully folks will enjoy having better units support and support for colors!

I really appreciate you taking the time to add these features and, even more, taking the time to deal with my slow responses.

**Comment by baxerus** [comment]
Hey Marty, everything is okay. I wasn't the fastest to answer either. Let's both be happy, that the plugin is now better than ever. We only need more people to use it 😃 Because it is a very good way to import 2D graphics into OpenSCAD (I think the best in the moment). It is also cool to have only one file in the end, without dependencies to a 2D vector file (SVG or other format) laying around somewhere!

By the way, OpenSCAD is experimenting with SVG import, but I do not know how good it is working.
